Transform nodes cleanup

BryceBubbles opened the pull request:
@HungryProton 
Hey, sorry if this is a huge pain...
I'm pretty new to the git workflow, so I'm not sure on the right way to do a bunch of things.

I've made quite a lot of changes here, so feel free to ask me to break it up into smaller pull requests!?

Is it better to merging any changes from the main develop branch into my branch before doing a pull request?

I started to do that, but then wasn't sure how to go about merging the conflicts (2 files that were renamed in upstream), and if I'm supposed to also commit the non-conflicting changes from upstream/develop as well (git messages seemed to imply that). 

It seems like there's a few different ways to approach this, so some direction on what you'd prefer would be awesome..
But also feel free to ditch this PR and tell me to go learn this stuff on my own :)

---
Anyway, in the PR I've:

- Made the node menu order alphabetically
- Tweaked display name and descriptions on transform nodes for consistency
- Added the missing single and combined transform nodes
- And lastly updated the display name for most other nodes

Lastly, lots of these node files should probably be renamed for consistency, but I wasn't sure if you'd be ok with me doing lots of renaming.

HungryProton commented:
Thanks for your contribution! I'll start the merge right away.

That's a huge PR though, it's usually better to split it in smaller ones (one per new feature). This way if only a part of the PR is problematic, the rest can still be merged in the meantime. Don't worry about this one though, I briefly looked through, it should be fine.

Usually you have to merge the target branch on your own before making a pull request. It's just a renamed file so I'll take care of it during the merge. Also, we're still in alpha and the next release will break a lot of things so I guess it's okay, but in the future, it would be nice to not change the slots id to avoid breaking backwards compatibility.

File naming is something that should be fixed too, along with the categories, you can give your input on #54 if you'd like.

BryceBubbles commented:
Sweet, yeah I'll keep PR's smaller and will pre-merge with target.
And PR to develop is best?

Yeah I'm pretty sure I didn't modify slot numbers (for that reason). I just changed names and added extra inputs on the end for some nodes. We really need that slot order decoupled from id feature soon.

Note: I also optimised the transform nodes a fair bit.

HungryProton commented:
Yes, PR to develop and once everything is stable enough for a while, I merge develop in master, make a proper release and a changelog.

Good catch on the `inside tree`, I didn't knew what caused this before. It happens when a script tries to modify the global_transform of a spatial that's not in a scene tree (and a lot of nodes are manipulated outside of it, `add_child` is called at the very end of the generation on what's passed to the `output` nodes). I should be able to track back the last bits of code that still produce this error now.
